src/copilot/context/contextCache.ts

```diff
@@ -296,8 +299,16 @@ export class ContextCache {
         }
         
         try {
+            // Fast check using document version first
             const document = await vscode.workspace.openTextDocument(uri);
-            const currentHash = crypto.createHash('md5').update(document.getText()).digest('hex');
+            if (entry.documentVersion !== undefined && document.version !== entry.documentVersion) {
+                return true;
+            }
+            
+            // If document version is the same or not available, check file stats
+            const stats = await vscode.workspace.fs.stat(uri);
+            const hashInput = `${document.version}-${stats.mtime}-${stats.size}`;
+            const currentHash = crypto.createHash('md5').update(hashInput).digest('hex');
```

Comment:
why not use the hashInput as key directly instead of md5?

Reply:
updated!